.github/workflows/main-cmake.yml

```diff
@@ -225,13 +258,7 @@ jobs:
       - name: CMake Run Tests
         run: ctest . --parallel 2 -C ${{ inputs.build_mode }} -V
         working-directory: ${{ runner.workspace }}/build
-        if: ${{ matrix.run_tests && (inputs.thread_safety != 'TS') }}
```

Comment:
What was this part changed for? Previously, this was to skip everything but the thread-safe tests for thread-safe builds.

Reply:
I added CI testing for the new concurrency configure option.

Reply:
Sure, I guess the question was more "why change the behavior for the existing thread-safe tests?" We don't necessarily need to run the whole test suite for thread-safe builds.

Reply:
Ah - well, I figured that there were going to be continuing changes to the threading and concurrency code, so it would be a good defensive move to run all the tests now, for both the threadsafe and concurrency builds.

If you think that's too much, I could revert the change for the threadsafe builds, and just leave the full testing enabled for the concurrency builds.

Reply:
That's basically what I was thinking, though it'd be good to hear from others. I'm just always cautious about adding too much to the CI testing. We're definitely at a point where we start to serialize things and bog down a bit due to the matrix of testing run for each PR/commit. At least until we stop testing Autotools.